torch/distributed/_tensor/dispatch.py

```diff
@@ -21,7 +21,12 @@
 from torch.distributed._tensor.random import is_rng_supported_mesh
 from torch.distributed._tensor.redistribute import redistribute_local_tensor
 from torch.distributed._tensor.sharding_prop import ShardingPropagator
-from torch.utils._pytree import tree_flatten, tree_unflatten
+
+try:
+    from torch.utils._cxx_pytree import tree_flatten, tree_unflatten
+except ImportError:
+    tree_flatten = torch.utils._pytree.tree_flatten  # type: ignore[assignment]
```

Comment:
curious, why the assignment version here instead of doing import like above

Reply:
I followed a similar approach taken in https://github.com/pytorch/pytorch/pull/109684, probably another import also works (in both case we need to ignore type annotation, lmk your preference :)